kit/src/main/java/com/oracle/javafx/scenebuilder/kit/editor/EditorPlatform.java

```diff
@@ -331,54 +344,44 @@ public static void open(String path) throws IOException {
+     * On Linux, it runs {@code 'xdg-open'}. On Mac, it runs {@code 'open'} and on Windows it runs {@code 'explorer /select'}.
      *
      * @param filePath path for the folder to be revealed
-     * @throws IOException if an error occurs
+     * @throws FileBrowserRevealException This exception allows to catch exits codes != 0 from the called process.
+     * @throws IOException General IOExceptions are thrown by Java System Call Processes in case of an error. 
      */
-    public static void revealInFileBrowser(File filePath) throws IOException {
+    public static void revealInFileBrowser(File filePath) throws IOException, FileBrowserRevealException {
         List<String> args = new ArrayList<>();
-        String path = filePath.toURI().toURL().toExternalForm();
+        String path = Paths.get(filePath.toURI()).normalize().toAbsolutePath().toString();
+        int exitCodeOk = 0;
         if (EditorPlatform.IS_MAC) {
             args.add("open"); //NOI18N
             args.add("-R"); //NOI18N
             args.add(path);
         } else if (EditorPlatform.IS_WINDOWS) {
             args.add("explorer"); //NOI18N
             args.add("/select," + path); //NOI18N
-        } else if (EditorPlatform.IS_LINUX) {
-            // nautilus does fine on Ubuntu, which is a Debian.
-            // I've no idea how it does with other Linux flavors.
-            args.add("nautilus"); //NOI18N
-            // The nautilus that comes with Ubuntu up to 11.04 included doesn't
-            // take a file path as parameter (you get an error popup), you must
-            // provide a dir path.
-            // Starting with Ubuntu 11.10 (the first based on kernel 3.x) a
-            // file path is well managed.
-            int osVersionNumerical = Integer.parseInt(System.getProperty("os.version").substring(0, 1)); //NOI18N
-            if (osVersionNumerical < 3) {
-                // Case Ubuntu 10.04 to 11.04: What you provide to nautilus is
-                // the name of the directory containing the file you want to see
-                // listed. See DTL-5384.
-                path = filePath.getAbsoluteFile().getParent();
-                if (path == null) {
-                    path = "."; //NOI18N
-                }
+            exitCodeOk = 1;
+        } else if (EditorPlatform.IS_LINUX) {           
+            args.add("xdg-open"); //NOI18N
+            path = filePath.getAbsoluteFile().getParent();
+            if (path == null) {
+                path = "."; //NOI18N
             }
             args.add(path);
         } else {
             // Not Supported
+            LOGGER.log(Level.SEVERE, "Unsupported operating system! Cannot reveal location {0} in file browser.", path);
         }
 
         if (!args.isEmpty()) {
-            executeDaemon(args, null);
+            executeDaemon(args, null, exitCodeOk);
```

Comment:
question: are we running this from the JavaFX thread? It should be done in a background thread if that was the case.

Reply:
I  think it does run within JavaFX thread, as with my experiments with blocking tasks, the UI did actually freeze. After a quick test using `System.out.println("CURR THREAD: " + Thread.currentThread().getName());` it turned out, that it is indeed the JavaFX Application Thread.

To solve this, instead of using `Platform.runLater()` I would use a `Task` so we can collect any exceptions.
What do you thing @jperedadnr ?